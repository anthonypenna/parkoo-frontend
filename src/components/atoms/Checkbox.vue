<template>
  <div class="checkbox">
    <label class="checkbox__label" :for="name">
      <slot name="label" />
    </label>
    <div class="checkbox__input">
      <input
        v-bind="$attrs"
        type="checkbox"
        :name="name"
        :checked="checked"
        @change="onChange"
      />
      <Checkmark v-show="checked" />
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import Checkmark from "@/components/atoms/Checkmark.vue";

export default Vue.extend({
  name: "Checkbox",
  components: { Checkmark },
  model: {
    prop: "checked",
    event: "change",
  },
  props: {
    name: {
      default: "",
      type: String,
    },
    checked: {
      default: false,
      type: Boolean,
    },
  },
  methods: {
    onChange(event: { target: HTMLInputElement }) {
      this.$emit("change", event.target.checked);
    },
  },
});
</script>

<style lang="scss" scoped>
.checkbox {
  width: min-content;
  display: flex;
  flex-direction: row-reverse;

  &__label {
    font-weight: bold;
  }

  &__input {
    position: relative;
    width: 20px;
    height: 20px;
    margin-right: 0.5rem;
    border: solid 2px #e4e4e4;
    border-radius: 5px;
  }

  input {
    position: absolute;
    top: -2.5px;
    left: -4px;
    width: 20px;
    height: 20px;
    opacity: 0;
  }
}
</style>
