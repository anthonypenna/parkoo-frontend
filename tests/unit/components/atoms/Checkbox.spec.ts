import { shallowMount } from '@vue/test-utils'
import Checkbox from '@/components/atoms/Checkbox.vue'

describe('<Checkbox />', () => {
  describe('when unchecked', () => {
    it('should render correctly', () => {
      const wrapper = shallowMount(Checkbox, {
        propsData: { name: 'foo' }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when checked', () => {
    it('should render correctly', () => {
      const wrapper = shallowMount(Checkbox, { propsData: { name: 'foo', checked: true } })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when clicked', () => {
    it('should emit a change event', () => {
      const wrapper = shallowMount(Checkbox)
      wrapper.find('input').trigger('change')
      expect(wrapper.emitted().change).toBeTruthy()
    })
  })
})
