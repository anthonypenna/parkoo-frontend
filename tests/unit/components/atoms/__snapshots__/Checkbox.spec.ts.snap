// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> when checked should render correctly 1`] = `
<div class="checkbox"><label for="foo" class="checkbox__label"></label>
  <div class="checkbox__input"><input type="checkbox" name="foo">
    <checkmark-stub></checkmark-stub>
  </div>
</div>
`;

exports[`<Checkbox /> when unchecked should render correctly 1`] = `
<div class="checkbox"><label for="foo" class="checkbox__label"></label>
  <div class="checkbox__input"><input type="checkbox" name="foo">
    <checkmark-stub style="display: none;"></checkmark-stub>
  </div>
</div>
`;
